app/hooks/use-media-query.tsx

```diff
@@ -0,0 +1,44 @@
+import {useEffect, useState} from 'react';
+
+function useMediaQuery(query: string): boolean {
+  const getMatches = (query: string): boolean => {
+    // Prevents SSR issues
+    if (typeof window !== 'undefined') {
+      return window.matchMedia(query).matches;
+    }
+    return false;
+  };
+
+  const [matches, setMatches] = useState<boolean>(false);
```

Comment:
Just FYI -> In this case of `useState` there's no need to specify the type. Because you're setting the default to a primitive type. 

You could do the following:

```typescript
  const [matches, setMatches] = useState(false);
```

Reply:
🪄